src/Texture2D.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+** EPITECH PROJECT, 2022
+** indieStudio
+** File description:
+** Texture2D
+*/
+
+
+#include <raylib.h>
+#include "Texture2D.hpp"
+
+namespace indie {
+
+Texture::Texture(const std::string &fileName)
+{
+    _texture = std::make_unique<Texture2D>(::LoadTexture(fileName.c_str()));
+    _isLoaded = true;
+}
+
+Texture::~Texture()
+{
+    if (_isLoaded)
+        UnloadTexture();
+}
+
+void Texture::LoadTexture(const std::string &fileName)
+{
+    if (_isLoaded)
+        UnloadTexture();
+    _texture = std::make_unique<Texture2D>(::LoadTexture(fileName.c_str()));
```

Comment:
Their is no check of LoadTexture return.
Eared that the Id of newly created texture2D may be set to an error value if loadTexture failed.